### PR TITLE
RMF: Rework write pipeline

### PR DESCRIPTION
**Pull request by drons**
* Parallelize data compression
* Add internal tile write-cache
* Add compressed overviews support.

### CODE REVIEW

**Comment by drons** [comment]
@rouault You restrict the parallel compression of the JPEG tiles in the [GTiff driver](https://github.com/OSGeo/gdal/blob/7f4574d89c07e14ef8fad0074480e4e6d1ffd4d7/gdal/frmts/gtiff/geotiff.cpp#L8462).
This restriction is only valid for GTiff, or should I also limit it?

File: gdal/frmts/rmf/rmfdataset.cpp [hunk 1]
```diff
@@ -2947,6 +3028,7 @@ void GDALRegister_RMF()
 "     <Value>RMF_DEM</Value>"
 "   </Option>"
 "   <Option name='JPEG_QUALITY' type='int' description='JPEG quality 1-100' default='75'/>"
+"   <Option name='NUM_THREADS' type='string' description='Number of worker threads for compression. Can be set to ALL_CPUS' default='1'/>"
```
Comment (rouault) [inline gdal/frmts/rmf/rmfdataset.cpp]: should be documented in the HTML

**Comment by drons** [comment]
> I think I tested it and found that there was no gain given that libjpegturbo compression is so fast and given the overhead of the multithreading extra steps

I'm test this option on my laptop (slow CPU, but NVME storage).
Speedup is ~2 times on 4 threads with internal libjpeg.

**Comment by drons** [comment]
@rouault Thanks!
But unfortunately, I'm find critical bug in my code. Fix here b786d99c33c4f603825131ade37c88b5f1976d72. 
Do I need to make a new PR, or you can 'cherry-pick' it?

**Comment by rouault** [comment]
https://github.com/OSGeo/gdal/commit/b786d99c33c4f603825131ade37c88b5f1976d72 cherry-picked and pushed

**Comment by drons** [comment]
Thanks!

**Pull request merged**